src/Raven.Server/Documents/Indexes/AbstractIndexCreateController.cs

```diff
@@ -223,6 +230,29 @@ private void ValidateAnalyzers(IndexDefinition definition)
             }
         }
     }
+    
+    private void ValidateConfiguration(IndexDefinition definition)
+    {
+        if (definition.Configuration == null)
+            return;
+        
+        foreach (var kvp in definition.Configuration)
+        {
+            if (_validConfigurationKeys.Contains(kvp.Key) == false)
+            {
+                throw new IndexCreationException($"Could not create index '{definition.Name}' because the configuration option key '{kvp.Key}' is not recognized");
+            }
+        }
+    }
+
+    private HashSet<string> GetValidConfigurationKeys()
+    {
+        var configurationProperties = ServerStore.Configuration.Indexing.GetConfigurationProperties();
+        
+        var validPerIndexConfigurationKeys = configurationProperties.SelectMany(configurationProperty => configurationProperty.ConfigurationEntryAttributes.Select(configurationEntryAttribute => configurationEntryAttribute.Key)).ToHashSet();
```

Comment:
I think I would prefer to use `AllConfigurationEntries` and maybe inside indexing use this static value, but just filter to indexing and create a hashset of those?

Reply:
Changed